mrjob/spark/harness.py

```diff
@@ -281,12 +284,22 @@ def make_mrc_job(mrc, step_num):
         if args.counter_output_dir is not None:
             counters = [ca.value for ca in counter_accumulators]
 
-            sc.parallelize(
-                [json.dumps(counters)],
-                numSlices=1
-            ).saveAsTextFile(
-                args.counter_output_dir
-            )
+            # If the given path is an s3 path, use s3.parallelize,
+            # otherwise just write them directly to the local dir
+            if is_uri(args.counter_output_dir):
+                sc.parallelize(
+                    [json.dumps(counters)],
+                    numSlices=1
+                ).saveAsTextFile(
+                    args.counter_output_dir
+                )
+                # Use regular python buildin file writer if the part-* file is not created
+            elif not glob(args.counter_output_dir + "/part-*"):
```

Comment:
I am not sure I understand why we need this check.  Are we just trying to avoid overwriting results from a previous run?  If so, that's now how we do it for the non-local counter_output_dir, so not sure why we'd need it here.  Or am I missing something?

Reply:
I will remove this check, this is not really required.